pkg/internal/apis/config/validate.go

```diff
@@ -192,6 +196,17 @@ func validatePortMappings(portMappings []PortMapping) error {
 	return nil
 }
 
+func validateDevices(devices []string) error {
+	for _, device := range devices {
+		device := strings.TrimSpace(device)
```

Comment:
You should be able to use `"github.com/container-orchestrated-devices/container-device-interface/pkg/parser"` and call `parser.IsQualifiedName(device)` here. This is used in https://github.com/docker/cli/pull/4084 and significantly reduces the dependencies pulled in. 

It should also be fine to allow `podman` and / or `docker` to perform this validation though.

Reply:
I'll give that a try.

Reply:
Please update to what Evan suggests to reduce the dependencies pulled in.

Reply:
@BenTheElder What do you think?

Reply:
@klueska I am using parser.ParseQualifiedName(device) instead of parser.ParseQualifiedName(device). The results are the same except you get the additional error message about why your device string is invalid. I think this is preferable. ParseQualifiedName() is called in IsQualifiedName().

This doesn't change the dependencies at all. There are tests on github.com/container-orchestrated-devices/container-device-interface/pkg/parser pulling in the additional packages. Maybe we should put the tests in a different package? @elezar wdyt?

Reply:
I can update the tests in that package to use a different import for comparison or to be in the `parser_test` package if require. As a matter of interest, which package that is equivalent to `require` would be preferred?

Reply:
I don't personally have a preference - I haven't found any with fewer dependencies. I was thinking use the `parser_test` package to keep things very lightweight.

Reply:
I have created https://github.com/container-orchestrated-devices/container-device-interface/pull/149 with an update to the test package. This should prevent `github.com/stretchr/testify/require` from getting pulled in.

Reply:
Sorry, a few things came up back in the real world and here with #3277 etc., the dependency discussion wound up forked in https://github.com/kubernetes-sigs/kind/pull/3290/files#r1304873501

I'm going to try to get https://github.com/kubernetes-sigs/kind/pull/3335 after which when this PR is rebased and `go mod tidy` it should be clearer what we're actually talking about for the new dependencies.

See linked comment for stance on deps.

Reply:
Currently for example we have a lot of CRI inspired types with in-tree dependency-free code instead, to avoid creating dep hell for our users. That may not be so reasonable here.